java/ql/src/semmle/code/java/dataflow/internal/ContainerFlow.qll

```diff
@@ -88,6 +91,290 @@ class ContainerType extends RefType {
   }
 }
 
+private class ContainerFlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "java.util;Map<>$Entry;true;getValue;;;MapValue of Argument[-1];ReturnValue;value",
+        "java.util;Map<>$Entry;true;setValue;;;MapValue of Argument[-1];ReturnValue;value",
+        "java.util;Map<>$Entry;true;setValue;;;Argument[0];MapValue of Argument[-1];value",
```

Comment:
Will this pair work as expected? (the return is the old value)

Reply:
Yes, when we're targeting an argument as the output then it actually means the corresponding post-update node. So there's no overlap between these two lines, and thus should work just fine (with the obvious exception that the update isn't reflected back to the underlying map, which is a lot harder to do, since that's moving into aliasing territory).